vendor/engines/secure_rooms/app/services/secure_rooms/access_handlers/occupancy_handler.rb

```diff
@@ -0,0 +1,67 @@
+module SecureRooms
+
+  module AccessHandlers
+
+    class OccupancyHandler
+
+      attr_reader :event
+
+      def self.process(event)
+        new(event).process
+      end
+
+      def initialize(event)
+        @event = event
+      end
+
+      def process
+        event_attribute =
+          if entering?
+            { entry_event: event }
+          else
+            { exit_event: event }
+          end
+
+        if current_occupant? && exiting?
+          existing_occupancy.update!(event_attribute)
+        elsif new_occupant? && entering?
+          new_occupancy.update!(event_attribute)
+        end
+
+        existing_occupancy || new_occupancy
+      end
+
+      def existing_occupancy
```

Comment:
it gets a little tricky because sometimes we want both the existing and the new (in one of the upcoming error cases) so I think we want them to be reference-able separately